lib/p2p/Pool.ts

```diff
@@ -81,7 +81,10 @@ class Pool extends EventEmitter {
       return;
     }
 
-    if (this.listenPort) {
+    if (this.server) {
+      await this.listen();
+      this.bindServer();
```

Comment:
This way we start listening for incoming connections before the external IP is appended. That could result in not advertising any addresses for a short period of time.

Reply:
True, I'll fetch our external IP first before we start listening.